Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -3002,17 +3013,22 @@ static FloatRect computeMarqueeSelectionRect(const WebCore::FloatPoint& point1,
     return FloatRect { [m_currentSelection boundsForPage:page.get()] }.contains(pagePoint);
 }
 
-FloatRect UnifiedPDFPlugin::rectForSelectionInPluginSpace(PDFSelection *selection) const
+FloatRect UnifiedPDFPlugin::rectForSelectionInMainFrameContentsSpace(PDFSelection *selection) const
 {
-    if (!selection || !selection.pages)
+    RefPtr mainFrameView = this->mainFrameView();
+    if (!mainFrameView)
         return { };
 
-    RetainPtr page = [selection.pages firstObject];
-    auto pageIndex = m_documentLayout.indexForPage(page);
-    if (!pageIndex)
+    if (!m_frame->coreLocalFrame())
+        return { };
+
+    RefPtr localFrameView = m_frame->coreLocalFrame()->view();
+    if (!localFrameView)
         return { };
 
-    return convertUp(CoordinateSpace::PDFPage, CoordinateSpace::Plugin, FloatRect { [selection boundsForPage:page.get()] }, *pageIndex);
+    auto rectForSelectionInRootView = this->rectForSelectionInRootView(selection);
+    auto rectForSelectionInContents = localFrameView->rootViewToContents(rectForSelectionInRootView);
+    return mainFrameView->windowToContents(localFrameView->contentsToWindow(roundedIntRect(rectForSelectionInContents)));
```

Comment:
At first glance, it's confusing that `windowToContents(contentsToWindow(X)) =/= X`!